### PR TITLE
fix TrailingWhitespace when using line breaks in macros arguments

### LINKED ISSUE #2916: internal error: left behind trailing whitespace
Source: https://github.com/bjorn3/rustc_codegen_cranelift/tree/rustfmt_error

```
$ cargo fmt
internal error: left behind trailing whitespace
   --> /home/bjorn/Documenten/rustc_codegen_cranelift/src/lib.rs:416
    |
416 |                     
    | ^^^^^^^^^^^^^^^^^^^^

warning: rustfmt may have failed to format. See previous 1 errors.
```

### PR DESCRIPTION
**Pull request by rchaser53**
fix #2916

### CODE REVIEW

File: src/rewrite.rs [hunk 1]
```diff
@@ -41,6 +42,7 @@ pub(crate) struct RewriteContext<'a> {
     pub(crate) macro_rewrite_failure: RefCell<bool>,
     pub(crate) report: FormatReport,
     pub(crate) skip_context: SkipContext,
+    pub(crate) skipped_range: Rc<RefCell<Vec<(usize, usize)>>>,
```
Comment (rchaser53) [inline src/rewrite.rs]: This is an important point in this PR. I want to use skipped_range in RewriteContext. It's used to resolve #3717. I think the ```skipped_range``` should include the failed format span or skipped span. This is related with #3761.

File: src/macros.rs [hunk 2]
```diff
@@ -181,6 +181,11 @@ fn return_macro_parse_failure_fallback(
         return trim_left_preserve_layout(context.snippet(span), indent, &context.config);
     }
 
+    context
+        .skipped_range
+        .borrow_mut()
+        .push((span.lo().to_usize(), span.hi().to_usize()));
```
Comment (topecongiro) [inline src/macros.rs]: The is adding byte positions of the span, whereas what we really need is the line numbers. We can get a line number of the given span via `SourceMap` from libsyntax.

Comment (rchaser53) [reply src/macros.rs]: Oh. I misunderstood. Thank you the review.

Comment (topecongiro) [reply src/macros.rs]: Thanks for the update!

**Pull request merged**